buildstockbatch/gcp/gcp.py

```diff
@@ -0,0 +1,1267 @@
+
+    def copy_files_at_cloud(self, files_to_copy):
+        """Implements :func:`DockerBase.copy_files_at_cloud`"""
+        logger.info("Copying weather files at Cloud Storage")
+        Parallel(n_jobs=-1, verbose=9)(
+            delayed(copy_GCS_file)(
+                self.gcs_bucket,
+                f"{self.gcs_prefix}/weather/{src}",
+                self.gcs_bucket,
+                f"{self.gcs_prefix}/weather/{dest}",
+            )
+            for src, dest in files_to_copy
+        )
+
+    def start_batch_job(self, batch_info):
+        """Implements :func:`DockerBase.start_batch_job`"""
+        # Make sure the default subnet for this region has access to Google APIs when external IP addresses are blocked.
+        subnet_client = compute_v1.SubnetworksClient()
+        op = subnet_client.set_private_ip_google_access(
+            project=self.gcp_project,
+            region=self.region,
+            subnetwork="default",
+            subnetworks_set_private_ip_google_access_request_resource=compute_v1.SubnetworksSetPrivateIpGoogleAccessRequest(  # noqa
+                private_ip_google_access=True
+            ),
+        )
+        op.result()
+        if op.error_code:
+            logger.error(
+                f"Error ({op.error_code}) updating subnet settings to allow private Google access: {op.error_message}"
+            )
+
+        # Define and run the GCP Batch job.
+        logger.info("Setting up GCP Batch job")
+        labels = {"bsb_job_identifier": self.job_identifier}
+
+        client = batch_v1.BatchServiceClient()
+
+        bsb_runnable = batch_v1.Runnable()
+        bsb_runnable.container = batch_v1.Runnable.Container()
+        bsb_runnable.container.image_uri = self.repository_uri + ":" + self.job_identifier
+        bsb_runnable.container.entrypoint = "/bin/sh"
+        bsb_runnable.container.options = "--rm"
+        bsb_runnable.labels = labels
+
+        # Pass environment variables to each task
+        environment = batch_v1.Environment()
+        # BATCH_TASK_INDEX and BATCH_TASK_COUNT env vars are automatically made available by GCP Batch.
+        environment.variables = {
+            "JOB_NAME": self.job_identifier,
+            "GCS_PREFIX": self.gcs_prefix,
+            "GCS_BUCKET": self.gcs_bucket,
+            "MISSING_ONLY": str(self.missing_only),
+        }
+        bsb_runnable.environment = environment
+
+        bsb_runnable.container.commands = ["-c", "python3 -m buildstockbatch.gcp.gcp"]
+
+        gcp_cfg = self.cfg["gcp"]
+        job_env_cfg = gcp_cfg.get("job_environment", {})
+        resources = batch_v1.ComputeResource(
+            cpu_milli=1000 * job_env_cfg.get("vcpus", 1),
+            memory_mib=job_env_cfg.get("memory_mib", 1024),
+            boot_disk_mib=job_env_cfg.get("boot_disk_mib", None),
+        )
+
+        # Give three minutes per simulation, plus ten minutes for job overhead
+        task_duration_secs = 60 * (10 + batch_info.n_sims_per_job * 3)
```

Comment:
This should be configurable in the project yaml file. Comstock simulations take on average of 10 minutes each. See the `hpc-spec` where we have to estimate a wall time submitting via sbatch.

Reply:
Add a config option for this.